sdks/python/src/opik/evaluation/metrics/heuristics/bleu.py

```diff
@@ -0,0 +1,218 @@
+    def score(
+        self, output: str, reference: Union[str, List[str]], **ignored_kwargs: Any
+    ) -> score_result.ScoreResult:
+        """
+        Computes a single-sentence BLEU score using nltk.translate.bleu_score.sentence_bleu.
+        If reference is a single string, it will be treated as one reference.
+        If reference is a list of strings, multiple references are used.
+        """
+        # 1) Handle empty candidate
+        if not output.strip():
+            return score_result.ScoreResult(
+                value=0.0,
+                name=self.name,
+                reason="Candidate is empty"
+            )
+
+        # 2) Process references
+        if isinstance(reference, str):
+            if not reference.strip():
+                return score_result.ScoreResult(
+                    value=0.0,
+                    name=self.name,
+                    reason="Reference is empty"
+                )
+            references = [reference.lower().split()]
+        else:
+            references = []
+            for ref in reference:
+                if not ref.strip():
+                    return score_result.ScoreResult(
+                        value=0.0,
+                        name=self.name,
+                        reason="Reference is empty"
+                    )
+                references.append(ref.lower().split())
+
+        candidate = output.lower().split()
+
+        # Truncate & normalize weights to the candidate length
+        used_weights = self._truncate_weights(len(candidate))
+
+        smoothing_func = self._get_smoothing_func()
+
+        try:
+            bleu_value = sentence_bleu(
+                references,
+                candidate,
+                weights=used_weights,
+                smoothing_function=smoothing_func
+            )
+        except ZeroDivisionError:
+            # edge case if references or candidate is basically empty after splitting
+            bleu_value = 0.0
+
+        return score_result.ScoreResult(
+            value=bleu_value,
+            name=self.name,
+            reason=f"Sentence-level BLEU (nltk, method={self.smoothing_method}): {bleu_value:.4f}",
+        )
+
+    ###########################################################################
+    # CORPUS-LEVEL BLEU
+    ###########################################################################
+    def score_corpus(
+        self,
+        outputs: List[str],
+        references_list: List[Union[str, List[str]]],
+        **ignored_kwargs: Any
+    ) -> score_result.ScoreResult:
+        """
+        Computes a corpus-level BLEU score using nltk.translate.bleu_score.corpus_bleu.
+        """
+
+        if len(outputs) != len(references_list):
+            return score_result.ScoreResult(
+                value=0.0,
+                name=self.name,
+                reason="Mismatch: number of candidates != number of references.",
+            )
+
+        all_candidates = []
```

Comment:
Please add type hints to these lists. It might be very helpful to understand the logic below (because of the fact that sometimes some values can be lists of strings and sometimes just strings).